### ISSUE TITLE
Not an issue, Suggestion:

### ISSUE THREAD

**Issue by GrenFlem** [sidezbros/double_hotbar#31]
The ability to automatically swap to the hotbar just by scrolling off the edge of the main one without swapping positions of them, effectively making the hotbar 18 slots

**Comment by The-Ice-Lord** [comment]
Yes, please! I was about to say the same thing, I don't want a new keybind or the need to press it all the time, I wish to just scroll through all the slots normally.

**Comment by codyisreallycool** [comment]
There used to be a mod named "Extended Hotbar" that did this. I used that mod up until Minecraft 1.21 was released, when it did not update to support the new version. I then used the unofficial pull until 1.21.2+ (iirc), when that pull also stopped working.

**Comment by sidezbros** [comment]
This is a client side mod intended to be usable on vanilla servers. That means it has to obey the server and the vanilla server does not allow the placement or use of any items/blocks not on the normal hotbar. This means there has to be a swap. Additionally, adding more features and injections into the game code increases the maintenance overhead of the mod for me and causes more crashes, bugs, and issues for all users. Adding the features requested in this issue would nearly double the codebase of the mod. I am aware of the other mods that have existed in the same area, but if this mod is to stay up to date and functional then I have to limit its size. 

In addition to the technical reasons, there are also design choices I made that I think make for a clear, easy to learn, and quick to master solution for using 18 slots. In early development I tested an “auto swap on scroll off the edge” function. It was clunky, confusing, and unappealing to me. My suggestion is to make more use of the “hold to swap a single stack” function or experiment with the option that flips the hotkey so you hold to swap the whole bar and tap to swap a single stack.

My apologies for the disappointing news, but I do appreciate the suggestions.

**Comment by GrenFlem** [comment]
There WOULD be swap, just not one that you can see, the bottom would display your normal hotbar, but when you scroll off of it, it displays the inventory, and the top one now displays the hotbar

**Comment by sidezbros** [comment]
Again, you can only interact with blocks and items on the bottom row of your inventory (the normal hotbar). Any visual trickery I do to give the appearance of the functionality you are asking for is just visual trickery. It gets very confusing for the player when you open your inventory and the two rows are not where you expect them to be because of what the HUD is displaying. This also ignores how additionally confusing it gets when you change which row in the inventory is the second hotbar. I have play tested this before and did not enjoy the experience. Thank you for your support and suggestion, but I will be closing the issue at this time.

**Comment by GrenFlem** [comment]
> Again, you can only interact with blocks and items on the bottom row of your inventory (the normal hotbar). Any visual trickery I do to give the appearance of the functionality you are asking for is just visual trickery. It gets very confusing for the player when you open your inventory and the two rows are not where you expect them to be because of what the HUD is displaying. This also ignores how additionally confusing it gets when you change which row in the inventory is the second hotbar. I have play tested this before and did not enjoy the experience. Thank you for your support and suggestion, but I will be closing the issue at this time.

I wouldnt really mind that thh hh iugh..
